player/lua/select.lua

```diff
@@ -345,6 +345,54 @@ mp.add_key_binding(nil, "select-audio-device", function ()
     })
 end)
 
+mp.add_key_binding(nil, "select-history", function ()
+    local watch_later_dir = mp.get_property("current-watch-later-dir")
+    local watch_later_files = {}
+
+    for i, file in ipairs(utils.readdir(watch_later_dir, "files") or {}) do
+        watch_later_files[i] = watch_later_dir .. "/" .. file
+    end
+
+    if #watch_later_files == 0 then
+        show_error("No watch later files found.")
+        return
+    end
+
+    local recent_files = {}
+    for _, watch_later_file in pairs(watch_later_files) do
+        local file_handle = io.open(watch_later_file)
+        if file_handle then
+            local line = file_handle:read()
+            if line and line ~= "# redirect entry" and line:find("^#") then
+                recent_files[#recent_files + 1] = {watch_later_file, line:sub(3)}
+            end
+            file_handle:close()
+        end
+    end
+
+    if #recent_files == 0 then
+        show_error("No watch later files with media paths found. " ..
+                   "Enable --write-filename-in-watch-later-config to select recent files.")
```

Comment:
This OSD message is too long. It should be concise and easy to read as it disappears fast.